### ISSUE TITLE
LD_LIBRARY_PATH is cleared when running with k8s

### ISSUE THREAD

**Issue by rulixxx** [nextflow-io/nextflow#3980]
## Bug report 

The container that I use defines an LD_LIBRARY_PATH in its Dockerfile. However when deployed with Nextflow using k8s executor it doesn't propagate into the pod resulting in execution errors due to the code not finding its required libraries.

### Steps to reproduce the problem

running this process in k8s:

```
process BAM_STATS {

    container "quay.io/wtsicgp/pcap-core:5.7.0"

    input:
        tuple val(meta), path(bam), path(bai)

    output:
        path "${meta.name}.samtools.flagstat.txt", emit: flagstat


    cpus 4
    memory 6.GB

    script:
        """
        samtools flagstat -@ $task.cpus $bam > ${meta.name}.samtools.flagstat.txt
        """
```

Results in the error:

`samtools: error while loading shared libraries: libhts.so.3: cannot open shared object file: No such file or directory`

Even though the Dockerfile for that image explicitly declares:  `ENV LD_LIBRARY_PATH $OPT/lib:$OPT/scramble/lib` so that doing : `docker run quay.io/wtsicgp/pcap-core:5.7.0 samtools` works as expected.

[Dockerfile](https://github.com/cancerit/PCAP-core/blob/develop/Dockerfile)

Works as expected with v22.XX of Nextflow but has become broken upon the 23.00 release

### Environment 

* Nextflow version:  23.05.0-edge
* Java version: 17.0.5
* Operating system: AWS

### Additional context

Using EKS/wave/fusion for execution

If I issue a `export` from within the process to inspect the environment variables, LD_LIBRARY_PATH is indeed missing while other env variables do make it correctly into the environment.

**Comment by pditommaso** [comment]
Please provide the `.nextflow.log` file and the `.command.*` files of the failing task

**Comment by rulixxx** [comment]
[nextflow.log](https://github.com/nextflow-io/nextflow/files/11601993/nextflow.log)

**Comment by rulixxx** [comment]
[command_files.tar.gz](https://github.com/nextflow-io/nextflow/files/11602159/command_files.tar.gz)

**Comment by pditommaso** [comment]
It's turn out this is a tricky problem related to Fusion. We are investigating for a solution

**Comment by jordeu** [comment]
This should be fixed with latest Fusion release. Latest release is automatically used when running with Nextflow edge version. @rulixxx can you give it a try and rerun it with Nextflow edge?

**Comment by rulixxx** [comment]
OK, will do. Thanks for looking into this...
________________________________

**Comment by rulixxx** [comment]
Hi Jordi,

Sorry for the late response...  I've just tried this and it does indeed work as expected with the edge release. Thanks for looking into this.
________________________________

**Comment by pditommaso** [comment]
I think this not yet fully addressed. @jordeu what's the current status of this issue?

**Comment by jordeu** [comment]
This was first fixed at v2.2.3, it was rollback at v2.2.6 due to a regression issue with old Kernels but it was fixed again on v2.2.7 in a way that keeps compatibility with old Kernels. So current status is: this issue is fixed when using a Kernel v4.18 or higher.

**Comment by pditommaso** [comment]
Fair enough. Closing this issue
